PF/Assignment1/BinarySearchTest.java

```diff
@@ -0,0 +1,73 @@
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+//test case class
+public class BinarySearchTest {
+	BinarySearch object = new BinarySearch();
+
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+	}
+
+	@AfterClass
+	public static void tearDownAfterClass() throws Exception {
+	}
+
+	@Before
+	public void setUp() throws Exception {
+	}
+
+	@After
+	public void tearDown() throws Exception {
+	}
+
+	@Test
+	// to test For Array With Identical Numbers
+	public void testForArrayWithIdenticalNumbers() {
+
+		int[] arr = { 1, 1, 1, 1, 1 };
+		int num = 1;
+		int index = object.binarySearch(num, 0, 4, arr);
+		int left = object.leftmostOccurance(arr, index);
+		int right = object.rightmostOccurance(arr, index);
+		assertEquals(2, index);
+		assertEquals(0, left);
```

Comment:
ACTION: Try to cover different cases where left most occurrence is  grater than 0